File: c++/leetcode/1046. Last Stone Weight.cpp
class Solution
{
public:
  int lastStoneWeight(vector<int> &stones)
  {
    while (stones.size() > 1)
    {
      const int sz = stones.size();
      sort(stones.begin(), stones.end());
      if (stones[sz - 1] == stones[sz - 2])
      {
        stones.pop_back();
        stones.pop_back();
        continue;
      }
      stones[sz - 2] = stones[sz - 1] - stones[sz - 2];
      stones.pop_back();
    }

    return stones.empty() ? 0 : stones.back();
  }
};

File: javascript/leetcode/1046. Last Stone Weight.js
/**
 * @param {number[]} stones
 * @return {number}
 */
var lastStoneWeight = function (stones) {
  while (stones.length > 1) {
    const sz = stones.length;
    stones.sort((a, b) => a - b);

    if (stones[sz - 1] === stones[sz - 2]) {
      stones.pop();
      stones.pop();
      continue;
    }
    stones[sz - 2] = stones[sz - 1] - stones[sz - 2];
    stones.pop();
  }
  return stones.length === 0 ? 0 : stones[0];
};
